metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -207,6 +230,10 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             "get_workunits_internal must be implemented if get_workunits is not overriden."
         )
 
+    # Not abstract for backwards compatibility
+    def get_config(self) -> Optional[ConfigModel]:
```

Comment:
not a huge fan of adding this to all sources

at a minimum, maybe we can have a default impl that looks at `self.config`, `self.source_config`, and a few other common names?

Reply:
Sure, that works. Ideally we standardize our sources and make them all have `self.config` so we don't even have this method (I guess that has to get enforced by a `Protocol`...), but this was my next option.